ArduCopter/radio.cpp

```diff
@@ -139,7 +139,7 @@ void Copter::set_throttle_and_failsafe(uint16_t throttle_pwm)
     if (throttle_pwm < (uint16_t)g.failsafe_throttle_value) {
 
         // if we are already in failsafe or motors not armed pass through throttle and exit
-        if (failsafe.radio || !(ap.rc_receiver_present || motors.armed())) {
+        if (failsafe.radio || !(ap.rc_receiver_present || hal.util->get_soft_arm_state() != AP_HAL::Util::SOFT_ARM_STATE_DISARMED)) {
```

Comment:
This could be made easier to read by removing the parentheses and making the new check an equal